interval/window_test.go

```diff
@@ -0,0 +1,1044 @@
+				{Start: mustTime("2020-06-01T00:00:00Z"), Stop: mustTime("2020-07-01T00:00:00Z")},
+				{Start: mustTime("2020-05-01T00:00:00Z"), Stop: mustTime("2020-06-01T00:00:00Z")},
+				{Start: mustTime("2020-04-01T00:00:00Z"), Stop: mustTime("2020-05-01T00:00:00Z")},
+				{Start: mustTime("2020-03-01T00:00:00Z"), Stop: mustTime("2020-04-01T00:00:00Z")},
+				{Start: mustTime("2020-02-01T00:00:00Z"), Stop: mustTime("2020-03-01T00:00:00Z")},
+				{Start: mustTime("2020-01-01T00:00:00Z"), Stop: mustTime("2020-02-01T00:00:00Z")},
+				{Start: mustTime("2019-12-01T00:00:00Z"), Stop: mustTime("2020-01-01T00:00:00Z")},
+			},
+		},
+		{
+			name: "end of month",
+			w: mustWindow(
+				values.ConvertDurationMonths(1),
+				values.ConvertDurationMonths(1),
+				values.ConvertDurationNsecs(-24*time.Hour),
+			),
+			t: mustTime("2020-10-01T00:00:00Z"),
+			want: []Bounds{
+				{Start: mustTime("2020-09-30T00:00:00Z"), Stop: mustTime("2020-10-31T00:00:00Z")},
+				{Start: mustTime("2020-08-31T00:00:00Z"), Stop: mustTime("2020-09-30T00:00:00Z")},
+				{Start: mustTime("2020-07-31T00:00:00Z"), Stop: mustTime("2020-08-31T00:00:00Z")},
+				{Start: mustTime("2020-06-30T00:00:00Z"), Stop: mustTime("2020-07-31T00:00:00Z")},
+				{Start: mustTime("2020-05-31T00:00:00Z"), Stop: mustTime("2020-06-30T00:00:00Z")},
+				{Start: mustTime("2020-04-30T00:00:00Z"), Stop: mustTime("2020-05-31T00:00:00Z")},
+				{Start: mustTime("2020-03-31T00:00:00Z"), Stop: mustTime("2020-04-30T00:00:00Z")},
+				{Start: mustTime("2020-02-29T00:00:00Z"), Stop: mustTime("2020-03-31T00:00:00Z")},
+				{Start: mustTime("2020-01-31T00:00:00Z"), Stop: mustTime("2020-02-29T00:00:00Z")},
+				{Start: mustTime("2019-12-31T00:00:00Z"), Stop: mustTime("2020-01-31T00:00:00Z")},
+			},
+		},
+		{
+			name: "far from bounds",
+			w: mustWindow(
+				values.ConvertDurationMonths(1),
+				values.ConvertDurationMonths(1),
+				values.ConvertDurationNsecs(0),
+			),
+			t: mustTime("2100-10-01T00:00:00Z"),
+			want: []Bounds{
+				{Start: mustTime("2100-10-01T00:00:00Z"), Stop: mustTime("2100-11-01T00:00:00Z")},
+				{Start: mustTime("2100-09-01T00:00:00Z"), Stop: mustTime("2100-10-01T00:00:00Z")},
+				{Start: mustTime("2100-08-01T00:00:00Z"), Stop: mustTime("2100-09-01T00:00:00Z")},
+				{Start: mustTime("2100-07-01T00:00:00Z"), Stop: mustTime("2100-08-01T00:00:00Z")},
+				{Start: mustTime("2100-06-01T00:00:00Z"), Stop: mustTime("2100-07-01T00:00:00Z")},
+				{Start: mustTime("2100-05-01T00:00:00Z"), Stop: mustTime("2100-06-01T00:00:00Z")},
+				{Start: mustTime("2100-04-01T00:00:00Z"), Stop: mustTime("2100-05-01T00:00:00Z")},
+				{Start: mustTime("2100-03-01T00:00:00Z"), Stop: mustTime("2100-04-01T00:00:00Z")},
+				{Start: mustTime("2100-02-01T00:00:00Z"), Stop: mustTime("2100-03-01T00:00:00Z")},
+				{Start: mustTime("2100-01-01T00:00:00Z"), Stop: mustTime("2100-02-01T00:00:00Z")},
+				{Start: mustTime("2099-12-01T00:00:00Z"), Stop: mustTime("2100-01-01T00:00:00Z")},
+			},
+		},
```

Comment:
@nathanielc ...and this test case to satisfy the request for window cases further away from the epoch. Is this what you were thinking of?